src/share/poudriere/jail.sh

```diff
@@ -580,13 +580,15 @@ install_from_vcs() {
 			msg_n "Updating the sources with ${METHOD}..."
 			${SVN_CMD} upgrade ${SRC_BASE} 2>/dev/null || :
 			${SVN_CMD} -q update -r ${TORELEASE:-head} ${SRC_BASE} || err 1 " fail"
+			${SVN_CMD} cleanup ${SRC_BASE} || err 1 " fail"
 			echo " done"
 			;;
 		git*)
 			${GIT_CMD} -C ${SRC_BASE} pull --rebase -q || err 1 " fail"
 			if [ -n "${TORELEASE}" ]; then
 				${GIT_CMD} checkout -q "${TORELEASE}" || err 1 " fail"
 			fi
+			(cd ${SRC_BASE} && ${GIT_CMD} gc) || err 1 " fail"
```

Comment:
I'm not sure this is necessary, git will automatically run gc when it needs to.

Reply:
Hmm, yes, it might be unnecessary.

Reply:
I keep wanting to avoid the kitchen sink of options in Poudriere. It's already bad enough. This could work in your update script right?
```
    poudriere jail -u
    zfs destroy snap@clean
    (cd & &git gc)
    zfs snapshot snap@clean
```
Someone else may want to just drop all commits before HEAD too to save space. Or they just don't care and want the update to be as fast as possible.

Reply:
Ah yes, this mini-script could work too. Such a cleanup should not take much time compared to buildworld, but the drop-everything-before-HEAD case is yet another option that one might or might not want.